packages/builder/src/pages/builder/app/[application]/settings/automations/index.svelte

```diff
@@ -38,8 +38,11 @@
   let loaded = false
   $: app = $appsStore.apps.find(app => $appStore.appId?.includes(app.appId))
   $: licensePlan = $auth.user?.license?.plan
-  $: page = $pageInfo.page
```

Comment:
Is it safe to remove this? We probably pulled this out so that it's stored in a primitive and can be properly compared. I'm worried that `fetchLogs(automationId, status, $pageInfo.page, timeRange)` could cause an endless loop as `$pageInfo.page` will re-trigger every time the `pageInfo` changes at all, which happens when we load data, which would trigger another refresh of data... etc.

I'm assuming this doesn't happen since you would have noticed it, but I would prefer to keep this as is and pass in the primitive `page` as the param to `fetchLogs` to keep reactivity to a minimum.

Reply:
I can leave it as it is, but if there is a loop it would happen like this as well, will it not? It would be a reactivity chain anyways

Reply:
Svelte uses shallow comparison (`==`) to determine if a value has changed or not, so if you have
```js
$: page = $pageInfo.page
$: console.log(page)
```
And the `pageInfo` store updates but the `page` (number) value stays the same, then the the log will not fire, because the primitive value stored in the first reactive statement will be equal when using shallow comparison. That's a benefit of using primitives for reactive statement values. If it was an object, then it would always retrigger no matter what.
```js
$: console.log($pageInfo.page)
```
Would trigger every time `pageInfo` updates even if no change is made to `page` directly, since it is observing the entire `pageInfo` store rather than just the primitive `page`.

It's a useful svelte trick to know.

Reply:
Cool, thx for the info! I did undo these changes :)